### DISCUSSION TITLE
Add mypy (or similar) to Actions?

### DISCUSSION THREAD

**Discussion by swamper123** [FreeOpcUa/opcua-asyncio#694] [Ideas]
Adding annotations and typing to the project is nice. But if we doesn't check if these types are correct or in collision with other stuff , it is just a "nice to have".
Sadly, Python itself doesn't check this kind of stuff, since it is an "optional" thing.

Mypy would check these parts and I would recommened to integrate it in the Actions. Alternatives are welcomened as well.

What do you think about it?

**Comment by swamper123** [comment]
The link to the mypy project: http://mypy-lang.org/ 

EDIT:

The Github Project: https://github.com/python/mypy

**Comment by xydan83** [comment]
By the way, we can also try using the full range of code checks and visualizations:
1. pylint - nice linter
2. isort - to sort imports
3. black - to improve the visualization of the code, although I don't really like it, I prefer to use formatting via PyCharm.

**Reply by swamper123** [reply]
I don't mind if pylint or flake8 would be used. Both are great imo.
After a quick research I don't know if isort is realy needed. Well it looks okay, but more as an optional tool. ^^
About black: I don't like it when tools can modify code. I prefere the "look at this"-principe with the programer in repsonse.

**Comment by oroulet** [comment]
I am not against mypy. anyone can make a PR I suppose. But we probably need some more typing before it gives any interesting result.
We already use flake8 I think and run pylint localy quite often.

I have really being considering using black on all code, but it requires one big commit to format everything first and I am bit afraid of that.

**Reply by swamper123** [reply]
It would be an interesting result later, indeed, but it is never to late for correct typing at the beginning. Otherwise we need more monster commits to fix `add here type checker` complains as well. ^^
